lib/srv/exec.go

```diff
@@ -436,13 +436,14 @@ func emitExecAuditEvent(ctx *ServerContext, cmd string, execErr error) {
 				scpEvent.Code = events.SCPDownloadCode
 			}
 		}
-		if err := ctx.srv.EmitAuditEvent(ctx.srv.Context(), scpEvent); err != nil {
```

Comment:
For now I think the best we can do is add a comment.

```
// use ctx.srv instead of ctx.session to emit the event to ensure that it gets recorded
```